accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockServiceTest.java

```diff
@@ -130,8 +130,53 @@ public void testBlockSizeAndIntervalWithMultipleInstances() {
         assertEquals(0, block1.getFirstValue());
         assertEquals(999, block1.getLastValue());
         ContiguousIdBlock block2 = service.reserveNewBlock(CATEGORY_ID_2, INSTANCE_ID_2);
-        assertEquals(3000, block2.getFirstValue());
-        assertEquals(3999, block2.getLastValue());
+        assertEquals(2000, block2.getFirstValue());
+        assertEquals(2999, block2.getLastValue());
         assertEquals(block2, repository.findFirstByCategoryIdOrderByLastValueDesc(CATEGORY_ID_2));
+
+        //Manually save a block of size 500, so for the current range only a block size of 500 reserved
+        repository.save(new ContiguousIdBlock(CATEGORY_ID_2, INSTANCE_ID, 4000, 500));
+        //Reserve a new block with size 1000
+        ContiguousIdBlock block3 = service.reserveNewBlock(CATEGORY_ID_2, INSTANCE_ID_2);
+
+        assertEquals(4500, block3.getFirstValue());
+        // the block is reserved only with size 500 the remainingSize in the range is 500 only
+        assertEquals(4999, block3.getLastValue());
+
+        // for remaining elements service would reserve new block interleaved by 1000.
+        ContiguousIdBlock block4 = service.reserveNewBlock(CATEGORY_ID_2, INSTANCE_ID_2);
+        assertEquals(6000, block4.getFirstValue());
+        // the block is reserved only with size full size 1000
+        assertEquals(6999, block4.getLastValue());
+    }
+
+    @Test
+    public void testNextBlock() {
+        // Reserving initial block
+        ContiguousIdBlock block1 = new ContiguousIdBlock(CATEGORY_ID,INSTANCE_ID,0,1000);
+        assertEquals(0, block1.getFirstValue());
+        assertEquals(999, block1.getLastValue());
+
+        ContiguousIdBlock block2 = block1.nextBlock(INSTANCE_ID, 2000, 0, 0);
+        assertEquals(1000, block2.getFirstValue()); // does not interleave as interleaveInterval = 0
+        assertEquals(2999, block2.getLastValue()); // as there is no interleaving any size can be reserved for a block
+
+        //Test different instance and different size
+        block2 = block1.nextBlock(INSTANCE_ID, 500, 1000, 0);
+        assertEquals(2000, block2.getFirstValue()); // interleaves as interleavingPoint is multiple of 1000
+        assertEquals(2499, block2.getLastValue());
+        //Reserving block with different instance and different size
+        ContiguousIdBlock block3 = block2.nextBlock(INSTANCE_ID_2, 1000, 1000, 0);
```

Comment:
how is it getting the next block after block2, but it returns a block that overlaps block2? if the data is independent of other variables, make it a separate test. if it is dependent, make the dependency clear and organised. here I can see how you overwrite block2 and block3, and it's not clear when it is important

Reply:
oh, I see, it doesn't overlap because the variable block2 was overwritten. exactly my second point

Reply:
split it in 09b8c87